### ISSUE TITLE
Warning (org-element): ‘org-element-at-point’ cannot be used in non-Org buffer

### ISSUE THREAD

**Issue by minad** [oantolin/embark#698]
When I use `embark-act` in the Agenda I get the following warning:

~~~
Warning (org-element): ‘org-element-at-point’ cannot be used in non-Org buffer #<buffer *Org Agenda*> (org-agenda-mode)
~~~ 

This seems to be a recent change, only introduced in development Org versions. This code in embark-org-target-element-context triggers the problem:

~~~ elisp
(org-element-lineage (org-element-context) embark-org--types t)
~~~

**Comment by minad** [comment]
See also https://github.com/alphapapa/org-web-tools/issues/56. I don't know if there is a good alternative for the agenda.

**Comment by minad** [comment]
Two possible "fixes":

1. Use only `(derived-mode 'org-mode)` in the target finder.
2. Let bind major mode as follows: `(let ((major-mode 'org-mode)) (org-element-lineage (org-element-context) embark-org--types t))`.

**Comment by minad** [comment]
Omar Antolín Camarena ***@***.***> writes:

> I think the best fix is probably your 1, use the target finder only in org-mode. I
> think 99% of the org targets I find useful in org-agenda buffers are headings and
> links, and both are found by other target finders. I don't think it would be much
> of a loss to have the element context target finder work only in org buffers.

Agree!

**Comment by minad** [comment]
Omar Antolín Camarena ***@***.***> writes:

> Closed #698 as completed via e0ee1c7.

Thanks!
